DEVELOPER_GUIDE.md

```diff
@@ -0,0 +1,135 @@
+
+```shell
+bundle exec jekyll spec-insert --watch
+```
+
+Depending on the text editor you are using to edit the markdown, you may need to manually reload the file from disk to see the changes applied by the plugin if the editor does not automatically reload the file periodically.
+
+The plugin will pull the newest OpenSearch API spec from its [repository](https://github.com/opensearch-project/opensearch-api-specification) if the spec file does not exist locally or if it is older than 24 hours. To tell the plugin to always pull the newest spec, you can add the `--refresh-spec` (or `-R`) flag to the command:
+
+```shell
+bundle exec jekyll spec-insert --refresh-spec
+```
+
+### Inserting query parameters
+
+To insert query parameters table of the, use the following snippet:
+
+```markdown
+<!-- spec_insert_start
+api: cat.indices
+component: query_parameters
+-->
+<!-- spec_insert_end -->
+```
+
+This will insert the query parameters of the `cat.indices` API into the `.md` file with three default columns: `Parameter`, `Type`, and `Description`. There are 5 columns that can be inserted: `Parameter`, `Type`, `Description`, `Required`, and `Default`. When `Required`/`Default` is not chosen, the info will be written in the `Description` column.
+
+You can customize the query parameters table with the following columns:
+
+- `Parameter`
+- `Type`
+- `Description`
+- `Required` 
+- `Default`
+
+ You can also customize this component with the following settings:
+ 
+- `include_global` (Boolean; default is `false`): Includes global query parameters in the table.
+- `include_deprecated` (Boolean; default is `true`): Includes deprecated parameters in the table.
+- `pretty` (Boolean; default is `false`): Renders the table in the pretty format instead of the compact format.
+
+You can customize the query parameters table with the following columns:
+
+```markdown
+<!-- spec_insert_start
+api: cat.indices
+component: query_parameters
+include_global: true
+include_deprecated: false
+pretty: true
+-->
+<!-- spec_insert_end -->
+```
+
+### Inserting path parameters
+
+To insert path parameters table of the `indices.create` API, use the following snippet:
+
+```markdown
+<!-- spec_insert_start
+api: indices.create
+component: path_parameters
+-->
+<!-- spec_insert_end -->
+```
+
+This table behaves the same as the query parameters table except that it does not accept the `include_global` argument.
+
+### Inserting paths and HTTP methods
+
+To insert paths and HTTP methods of the `search` API, use the following snippet:
+
+```markdown
+<!-- spec_insert_start
+api: search
+component: paths_and_http_methods
+-->
+<!-- spec_insert_end -->
+```
+
+### Ignoring files and folders
+
+The `spec-insert` plugin ignores all files and folders listed in the [./_config.yml#exclude](./_config.yml) list, which is also the list of files and folders that Jekyll ignores.
+
+### CI/CD
+
```

Comment:
Below: Is the link complete?

Reply:
This link will work once we have this merged in.